exercises/allergies/.gitignore

```diff
@@ -0,0 +1,7 @@
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
```

Comment:
And as a corollary, let's say that we do start ignoring Cargo.lock (as it appears we should).

Should we thus remove all Cargo.lock files from the repository in this case?
(That can go in a separate commit and can be a follow-up PR)